### PR TITLE
beluga: 20180403 -> unstable-2020-03-11

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Motivation for this change

Update a package and perform some necesary dependency updates.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/dune-build-info/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  inherit (dune_2) src version;
+
+  useDune2 = true;
+
+  dontAddPrefix = true;
+
+  meta = with lib; {
+    inherit (dune_2.meta) homepage;
+    description = "Embed build information inside executables";
+    maintainers = [ maintainers.bcdarwin ];
```
Comment (JeffLabonte) [inline pkgs/development/ocaml-modules/dune-build-info/default.nix:14]: You could write  `with maintainers; [ bcdarwin ];

File: pkgs/development/ocaml-modules/sedlex/default.nix [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
-
-  buildFlags = [ "all" "opt" ];
-
-  createFindlibDestdir = true;
-
-  dontStrip = true;
+  doCheck = true;
 
   meta = {
     homepage = https://github.com/ocaml-community/sedlex;
```
Comment (JeffLabonte) [inline pkgs/development/ocaml-modules/sedlex/default.nix:27]: You need to add `"` to the homepage. Please :)

**Review by JeffLabonte — CHANGES_REQUESTED** [review]
Beluge builds but there is a package that doesn't: 

```bash
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/jflabonte/.cache/nixpkgs-review/pr-84535-1/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
builder for '/nix/store/vibsabsv18yqm7wm0jlgmjcjf7r4k0k4-ocaml4.07.1-iri-0.4.0.drv' failed with exit code 2; last 10 log lines:
  building
  build flags: SHELL=/nix/store/ffli6m23501dkiznwlkf6n4xvrj02snr-bash-4.4-p23/bin/bash
  ocamlfind ocamlc -c -package sedlex,uutf,uunf.string -annot -bin-annot -safe-string -g iri_types.mli
  ocamlfind ocamlc -c -package sedlex,uutf,uunf.string -annot -bin-annot -safe-string -g iri_lexer.mli
  ocamlfind ocamlc -c -package sedlex,uutf,uunf.string -annot -bin-annot -safe-string -g iri.mli
  ocamlfind ocamlc -c -package sedlex,uutf,uunf.string -annot -bin-annot -safe-string -g iri_types.ml
  ocamlfind ocamlc -c -package sedlex,uutf,uunf.string -annot -bin-annot -safe-string -g iri_lexer.ml
  File "iri_lexer.ml", line 28, characters 14-27:
  Error: Uninterpreted extension 'sedlex.regexp'.
  make: *** [Makefile:69: iri_lexer.cmo] Error 2
cannot build derivation '/nix/store/ih4lpbqkkj7lnycj97h429mgh3dfhs78-ocaml4.07.1-xtmpl-0.17.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x4113ahjknniq7pnwmhkzv08pp3hkwb2-higlo-0.6.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/aqal70g25991a94x573385bxhlyx1ddq-stog-0.18.0.drv': 2 dependencies couldn't be built
[1/0/2 built (1 failed)] building ocaml4.07.1-beluga-unstable-20200311 (checkPh[1/0/2 built (1 failed)] building ocaml4.07.1-beluga-unstable-20200311 (fixupPhcannot build derivation '/nix/store/qq2qrql10v195krcxazwl3wzrf12pa29-env.drv': 1 dependencies couldn't be built
[1 built (1 failed)]
error: build of '/nix/store/qq2qrql10v195krcxazwl3wzrf12pa29-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/84535
1 package failed to build:
stog

1 package built:
beluga
```

File: pkgs/development/ocaml-modules/sedlex/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/ocaml-modules/sedlex/default.nix]: Done.

**Comment by bcdarwin** [comment]
Hmm, I thought `ocamlPackages.iri` (hence `ocamlPackages.xtmpl` and hence `stog`) were previously broken but that doesn't seem to be the case.

I guess the best solution here is to add a `sedlex/1.x` expression?

**Comment by bcdarwin** [comment]
@JeffLabonte -- updated the PR so as not to touch `sedlex`, fixing the build.

**Review by JeffLabonte — APPROVED** [review]
LGTM

File: pkgs/applications/science/logic/beluga/default.nix [hunk 3]
```diff
@@ -1,32 +1,38 @@
-{ stdenv, fetchFromGitHub, ocamlPackages, omake }:
+{ stdenv, fetchFromGitHub, ocamlPackages, rsync }:
 
-stdenv.mkDerivation {
-  name = "beluga-20180403";
+ocamlPackages.buildDunePackage {
+  pname = "beluga";
+  version = "unstable-20200311";
```
Comment (bhipple) [inline pkgs/applications/science/logic/beluga/default.nix:5]:
```suggestion
  version = "unstable-2020-03-11";
```
Generally we put dashes to make this readable.

Comment (bcdarwin) [reply pkgs/applications/science/logic/beluga/default.nix]: @bhipple -- sure, updated expression/commit/PR with this convention.

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 84535` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>49 packages marked as broken and skipped:</summary>
<br>- clang-sierraHack-stdenv
<br>- crossLibcStdenv
<br>- digitalbitbox
<br>- gdata-sharp
<br>- iterm2
<br>- linuxPackages.sch_cake
<br>- linuxPackages_4_14.sch_cake
<br>- linuxPackages_4_4.evdi
<br>- linuxPackages_4_4.sysdig
<br>- linuxPackages_4_4.zenpower
<br>- linuxPackages_4_9.sysdig
<br>- linuxPackages_4_9.zenpower
<br>- linuxPackages_5_6.sch_cake
<br>- linuxPackages_hardkernel_4_14.bcc
<br>- linuxPackages_hardkernel_4_14.bpftrace
<br>- linuxPackages_hardkernel_4_14.can-isotp
<br>- linuxPackages_hardkernel_4_14.chipsec
<br>- linuxPackages_hardkernel_4_14.digimend
<br>- linuxPackages_hardkernel_4_14.evdi
<br>- linuxPackages_hardkernel_4_14.mba6x_bl
<br>- linuxPackages_hardkernel_4_14.nvidia_x11
<br>- linuxPackages_hardkernel_4_14.nvidia_x11_legacy390
<br>- linuxPackages_hardkernel_4_14.prl-tools
<br>- linuxPackages_hardkernel_4_14.r8125
<br>- linuxPackages_hardkernel_4_14.r8168
<br>- linuxPackages_hardkernel_4_14.rtl8814au
<br>- linuxPackages_hardkernel_4_14.rtl8821au
<br>- linuxPackages_hardkernel_4_14.rtl8821ce
<br>- linuxPackages_hardkernel_4_14.rtlwifi_new
<br>- linuxPackages_hardkernel_4_14.sysdig
<br>- linuxPackages_hardkernel_4_14.v4l2loopback
<br>- linuxPackages_hardkernel_4_14.vhba
<br>- linuxPackages_hardkernel_4_14.wireguard
<br>- linuxPackages_hardkernel_4_14.zenpower
<br>- linuxPackages_latest_hardened.sch_cake
<br>- linuxPackages_xen_dom0_hardened.sch_cake
<br>- muslCross
<br>- newlibCross
<br>- ocamlPackages.stog
<br>- octoprint
<br>- php74Extensions.couchbase
<br>- python27Packages.flitBuildHook
<br>- python27Packages.libmodulemd
<br>- python37Packages.nixpart
<br>- python37Packages.notify
<br>- python38Packages.libselinux
<br>- python38Packages.nixpart
<br>- python38Packages.poezio
<br>- qes
</details>
<details>
  <summary>5 packages built:</summary>
<br>- beluga
<br>- flow
<br>- ocamlPackages.dune-build-info
<br>- ocamlPackages.linenoise
<br>- ocamlPackages.sedlex_2
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**